crates/router/src/db/configs.rs

```diff
@@ -69,14 +66,11 @@ impl ConfigInterface for Store {
             .await
             .map_err(|error| report!(errors::StorageError::from(error)))?;
 
-        self.get_redis_conn()
-            .map_err(Into::<errors::StorageError>::into)?
-            .publish(
-                cache::IMC_INVALIDATION_CHANNEL,
-                CacheKind::Config((&inserted.key).into()),
-            )
-            .await
-            .map_err(Into::<errors::StorageError>::into)?;
+        cache::publish_into_redact_channel(
+            self.get_cache_store().as_ref(),
+            [CacheKind::Config((&inserted.key).into())],
+        )
+        .await?;
```

Comment:
Isn't there a possibility that a stale value would be left in Redis without being deleted? Would calling `redact_from_redis_and_publish()` help?